pyam/_aggregate.py

```diff
@@ -51,6 +53,39 @@ def _aggregate(df, variable, components=None, method=np.sum):
     return _group_and_agg(_df, [], method)
 
 
+def _aggregate_recursive(df, variable, components=None, method=np.sum):
+    _dfout = None
+    _df = df.copy()
+
+    mapping = []
+
+    for d in reversed(range(1, max(find_depth(df.data.variable)) + 1)):
+        depth = find_depth(df.data.variable)
+        var_list = (
+            df.data.variable[[i == d for i in depth]]
+            .unique()
+        )
+        vars_up = pd.Series([reduce_hierarchy(i, -1) for i in var_list]).unique()
+
+        if [i for i, entr in enumerate(vars_up) if entr.startswith(variable)]:
+            for v in vars_up:
+                mapping.append(v)
+
+    mapping = sorted(set(mapping))
+    mapping.reverse()
```

Comment:
The great thing about Python - you can almost read the code "reversed sorted set mapping"...
```suggestion
    mapping = reversed(sorted(set(mapping)))
```